packages/dcos-integration-test/extra/test_endpoints.py

```diff
@@ -76,10 +77,15 @@ def test_if_uiconfig_is_available(cluster):
 
 
 def test_if_dcos_history_service_is_up(cluster):
-    r = cluster.get('/dcos-history-service/ping')
 
-    assert r.status_code == 200
-    assert 'pong' == r.text
+    @retry(stop_max_delay=20000, wait_fixed=500)
```

Comment:
I get that this _can_ hit other masters, but it does so in a very liberal way:
(1) Retries for any state of request. We would like to fail if we hit a connection error
(2) I don't like retrying in tests on something we expect to be stable once the cluster is up; if history service is not up everywhere when this tests starts, then perhaps we should make a change to the wait_for_history_service up
(3) if we really want to validate that all masters have history service running, then we should make sure that we can get cluster.len(master)*FACTOR successful GET's in a row to be reasonably sure we actually hit something else.